pkg/reconciler/testing/topic.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"context"
+	"knative.dev/pkg/apis"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/GoogleCloudPlatform/cloud-run-events/pkg/apis/pubsub/v1alpha1"
+)
+
+// TopicOption enables further configuration of a Topic.
+type TopicOption func(*v1alpha1.Topic)
+
+// NewTopic creates a Topic with TopicOptions
+func NewTopic(name, namespace string, so ...TopicOption) *v1alpha1.Topic {
+	s := &v1alpha1.Topic{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	for _, opt := range so {
+		opt(s)
+	}
+	s.SetDefaults(context.Background())
+	return s
+}
+
+func WithTopicUID(uid types.UID) TopicOption {
+	return func(s *v1alpha1.Topic) {
+		s.UID = uid
+	}
+}
+
+// WithInitTopicConditions initializes the Topics's conditions.
+func WithInitTopicConditions(s *v1alpha1.Topic) {
+	s.Status.InitializeConditions()
+}
+
+func WithTopicTopic(topicID string) TopicOption {
```

Comment:
:smile: Should this be `WithTopicTopic` or `WithTopicTopicID`?

Reply:
topic == topicID for better or worse.

Reply:
Worse :) Make this `WithTopicID` so it doesn't stutter.

Reply:
`WithTopicID` will conflict with `WithChannelTopicID` and `WithPullSubscriptionTopic`